Tests/OpenAPIRuntimeTests/Interface/Test_ErrorHandlingMiddleware.swift

```diff
@@ -0,0 +1,144 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftOpenAPIGenerator open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftOpenAPIGenerator project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftOpenAPIGenerator project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import HTTPTypes
+
+import XCTest
+@_spi(Generated) @testable import OpenAPIRuntime
+
+
```

Comment:
I have added tests for testing the new middleware in isolation. 
I thought of adding UniversalServerTests with a mock handler and this error middleware. But that becomes more of an integration test. Let me know if you prefer them to be added.

Reply:
Adding these tests are great. IMO we don't need an integration test for this.